utils/generate-catalog.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# Generate OLM catalog file
+
+set -euo pipefail
+
+### CONSTANTS
+# Used as well in the subscription object
+DEFAULT_CHANNEL=preview
+###
+
+OPM="${1?:Error \$OPM not set. Bye}"
+YQ="${2?:Error \$YQ not set. Bye}"
+BUNDLE_IMG="${3?:Error \$BUNDLE_IMG not set. Bye}"
+CATALOG_FILE="${4?:Error \$CATALOG_FILE not set. Bye}"
+CHANNELS="${5:-$DEFAULT_CHANNEL}"
+
+CATALOG_FILE_BASEDIR="$( cd "$( dirname "$(realpath ${CATALOG_FILE})" )" && pwd )"
+CATALOG_BASEDIR="$( cd "$( dirname "$(realpath ${CATALOG_FILE_BASEDIR})" )" && pwd )"
```

Comment:
These are failing locally for me, returning:
```
CATALOG_FILE_BASEDIR: project path
CATALOG_BASEDIR: project path..
```

thus causing:

```
Error: open /Users/guilherme.cassolato/Workspace/go/src/github.com/kuadrant/authorino-operator-channel-entry.yaml: no such file or directory
make[1]: *** [/Users/guilherme.cassolato/Workspace/go/src/github.com/kuadrant/authorino-operator/catalog/authorino-operator-catalog/operator.yaml] Error 1
make: *** [catalog] Error 2
```

Reply:
Without `realpath`, it works:

```suggestion
CATALOG_FILE_BASEDIR="$( cd "$( dirname ${CATALOG_FILE} )" && pwd )"
CATALOG_BASEDIR="$( cd "$( dirname ${CATALOG_FILE_BASEDIR} )" && pwd )"
```